gateway-eth-ts/test/integration/GatewayTs.test.ts

```diff
@@ -0,0 +1,238 @@
+import {
+  BaseProvider,
+  getDefaultProvider,
+  Network,
+} from "@ethersproject/providers";
+import { BigNumber, Wallet } from "ethers";
+import { TokenData, TokenState } from "../../src/utils";
+import * as assert from "assert";
+import * as dotenv from "dotenv";
+import { GatewayTs } from "../../src/service/GatewayTs";
+import {
+  TEST_GATEWAY_TOKEN_ADDRESS,
+} from "../../src/service/testUtils";
+import { GatewayNetwork, GatewayNetwork__factory } from "../../src/contracts/typechain-types";
+import { BNB_TESTNET_CONTRACT_ADDRESSES, deployerWallet, gatekeeperOneTestnetWallet, gatekeeperTwoTestnetWallet, initTestNetwork, testNetworkName, userOneWallet } from "../utils";
+dotenv.config();
+
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+describe("GatewayTS", function () {
+  this.timeout(5_000);
+  let gateway: GatewayTs;
+  let provider: BaseProvider;
+  let network: Network;
+  let gatekeeper: Wallet;
+  let gatewayNetworkContract: GatewayNetwork;
+
+  let testNetworkId: bigint;
+
+  let sampleWalletAddress: string
+
+  before("Initialize GatewayTS class", async function () {
+    this.timeout(10000);
+    provider = getDefaultProvider("http://localhost:8545");
+    network = await provider.getNetwork();
+    gatekeeper = gatekeeperOneTestnetWallet.connect(provider);
+
+    gatewayNetworkContract = GatewayNetwork__factory.connect(BNB_TESTNET_CONTRACT_ADDRESSES.gatewayNetwork, gatekeeper);
+
+    await initTestNetwork(gatewayNetworkContract, gatekeeper);
+
+    testNetworkId = (await gatewayNetworkContract.getNetworkId(testNetworkName)).toBigInt();
+
+    sampleWalletAddress = Wallet.createRandom().address;
+
+    gateway = new GatewayTs(
+      gatekeeper,
+      BNB_TESTNET_CONTRACT_ADDRESSES.gatewayToken
+    );
+  });
+
+  it("should issue a token", async () => {
+    const randomAddress = Wallet.createRandom().address;
+    await (await gateway.issue(randomAddress, testNetworkId, 0, 0, {feeSender: sampleWalletAddress, feeRecipient: gatekeeper.address})).wait();
+
+    const token = await gateway.getFirstTokenOnNetwork(
+      randomAddress,
+      testNetworkId
+    );
+
+    assert.equal(token!.owner, randomAddress);
+    assert.equal(token!.state, TokenState.ACTIVE);
+  }).timeout(15000);
+
+  it.skip("should tolerate multiple tokens", async () => {
```

Comment:
Should any of these skips be removed / fixed?

Reply:
I have a follow up ticket for working on these.Don't want to remove them because they are valuable test.